PROTO_tests/tests/dataframe_test.py

```diff
@@ -0,0 +1,704 @@
+        df = ak.DataFrame({"col_A": ak.array([7, 3]), "col_B": ak.array([1, 9])})
+
+        # test against pdarray
+        test_df = df.isin(ak.array([0, 1]))
+        assert test_df["col_A"].to_list() == [False, False]
+        assert test_df["col_B"].to_list() == [True, False]
+
+        # Test against dict
+        test_df = df.isin({"col_A": ak.array([0, 3])})
+        assert test_df["col_A"].to_list() == [False, True]
+        assert test_df["col_B"].to_list() == [False, False]
+
+        # test against series
+        i = ak.Index(ak.arange(2))
+        s = ak.Series(data=ak.array([3, 9]), index=i.index)
+        test_df = df.isin(s)
+        assert test_df["col_A"].to_list() == [False, False]
+        assert test_df["col_B"].to_list() == [False, True]
+
+        # test against another dataframe
+        other_df = ak.DataFrame({"col_A": ak.array([7, 3], dtype=ak.bigint), "col_C": ak.array([0, 9])})
+        test_df = df.isin(other_df)
+        assert test_df["col_A"].to_list() == [True, True]
+        assert test_df["col_B"].to_list() == [False, False]
+
+    def test_multiindex_compat(self):
+        # Added for testing Issue #1505
+        df = ak.DataFrame({"a": ak.arange(10), "b": ak.arange(10), "c": ak.arange(10)})
+        df.groupby(["a", "b"]).sum("c")
+
+    def test_uint_greediness(self):
+        # default to uint when all supportedInt and any value > 2**63
+        # to avoid loss of precision see (#1983)
+        df = pd.DataFrame({"Test": [2**64 - 1, 0]})
+        assert df["Test"].dtype == ak.uint64
+
+    def test_head_tail_resetting_index(self):
+        # Test that issue #2183 is resolved
+        df = ak.DataFrame({"cnt": ak.arange(65)})
+        # Note we have to call __repr__ to trigger head_tail_server call
+
+        bool_idx = df[df["cnt"] > 3]
+        bool_idx.__repr__()
+        assert bool_idx.index.index.to_list() == list(range(4, 65))
+
+        slice_idx = df[:]
+        slice_idx.__repr__()
+        assert slice_idx.index.index.to_list() == list(range(65))
+
+        # verify it persists non-int Index
+        idx = ak.concatenate([ak.zeros(5, bool), ak.ones(60, bool)])
+        df = ak.DataFrame({"cnt": ak.arange(65)}, index=idx)
+
+        bool_idx = df[df["cnt"] > 3]
+        bool_idx.__repr__()
+        # the new index is first False and rest True (because we lose first 4),
+        # so equivalent to arange(61, bool)
+        assert bool_idx.index.index.to_list() == ak.arange(61, dtype=bool).to_list()
+
+        slice_idx = df[:]
+        slice_idx.__repr__()
+        assert slice_idx.index.index.to_list() == idx.to_list()
+
+    def test_ipv4_columns(self):
+        # test with single IPv4 column
+        df = ak.DataFrame({"a": ak.arange(10), "b": ak.IPv4(ak.arange(10))})
+        with tempfile.TemporaryDirectory(dir=TestDataFrame.df_test_base_tmp) as tmp_dirname:
+            fname = tmp_dirname + "/ipv4_df"
+            df.to_parquet(fname)
+
+            data = ak.read(fname + "*")
```

Comment:
since we're not including `test_save` and that's gonna be handled by the `io_test`, i feel like it would make sense for this to be handled in that test file as well

if we decide to do that we can also drop the 
```python
    df_test_base_tmp = "{}/df_test".format(os.getcwd())
    io_util.get_directory(df_test_base_tmp)
```
at the top

Reply:
Did we decide on keeping `test_ipv4_columns` here or moving it to `io_tests`?

Reply:
No we are moving it, my brain equated that to nothing needs to be done for some reason. I've updated this.